### DISCUSSION TITLE
Performance Rating results with different gcc option

### DISCUSSION THREAD

**Discussion by defdefred** [VendelinSlezak/BleskOS#74] [Ideas]
Could be nice to check if compiling option and proper target CPU options are making BleskOS faster or not:-)
https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html
https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html

**Comment by VendelinSlezak** [comment]
I am sure there are some possible optimizations, however when I try another -O option than -O0 it always broke BleskOS. I am not familiar how this whole process works and how would these options affect code from functionality side, so I do not really know how to use them. If you know more, you are welcome to try and experiment with them :)

**Comment by defdefred** [comment]
Did you try to change the target cpu too?
I guess that the misuse of dword_t instead of pointer could make the optimizer to misbehave...

**Comment by VendelinSlezak** [comment]
I invested some time into this, and I fixed what was broking optimization and according to performance rating it speeded up performance about unbelievable 55%! And image is smaller about 100 KB, so it is really two wins at once.

![No_optimization](https://github.com/VendelinSlezak/BleskOS/assets/43180618/2ab4358b-c3b7-4b3c-a863-df84a8dffee7)

![Optimization_s](https://github.com/VendelinSlezak/BleskOS/assets/43180618/9fc2487f-8f47-4eda-b879-b1968fc75a7b)

**Comment by defdefred** [comment]
What the point! Make more with the same hardware, I like it!
And still with basic x86 target cpu, or you tried -march=pentium4?

**Comment by VendelinSlezak** [comment]
I think default generic option is best, because BleskOS targets all types of CPU. And I tried some options, but it do not make any real difference.

**Comment by defdefred** [comment]
Must be a trick, because with -march=pentium3 gcc can use SIMD instruction (SSE, MMX) and it is a real game changer. Pentium3 is already +20y architecture :-)

What is the oldest computosaure you test BleskOS on?

**Comment by VendelinSlezak** [comment]
> -march=pentium3

I tried it, but file was only about 400 bytes smaller and when I tried to run Performance rating in QEMU, it throwed "invalid opcode" ISR error, so it is not usable.

> What is the oldest computosaure you test BleskOS on?

I have Compaq Armada 1750. It is about 20 years old. In this year it is really looks like something taken out from museum :)

**Comment by defdefred** [comment]
Did you add the cpu target in qemu?
```
qemu-system-i386 --cpu pentium3
```
I read that SSE has to be enable by the OS... https://wiki.osdev.org/SSE
So more code is needed and maybe VARIABLE to be defined to activate SSE or not if older cpu tested as your pentium2 (MMX ok but not SSE).
